setup.py

```diff
@@ -83,37 +83,43 @@
     # for better multiprocessing
     "multiprocess",
     # to get metadata of optional dependencies such as torch or tensorflow for Python versions that don't have it
-    "importlib_metadata;python_version<'3.8'"
+    "importlib_metadata;python_version<'3.8'",
+    # for saving datsets to local
+    "fsspec",
```

Comment:
We need fsspec in the requirements just for the local filesystem right ?
Maybe we can simply have our own LocalFileSystem from [their implementation](https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/implementations/local.html) instead in `datasets.filesystem. LocalFileSystem`

Reply:
Yes, we need `fsspec` for the local filesystem. So would copy their implementation instead of installing `fsspec` i can give it a try later.

Reply:
sadly not possible since `Localfilesystem` is having `AbstractFileSystem` as dependencies, which has several more dependencies. 
https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/spec.html#AbstractFileSystem

Reply:
We've taken a look with @philschmid and it's not just a class to copy so we'll need to have fsspec as a hard dependency